contracts/CourtRoom.sol

```diff
@@ -7,179 +7,160 @@ import "./ensabstract.sol";
+		OpenCases[_id].id = 0x0;
+		OpenCases[_id].plaintiff = 0;
+		OpenCases[_id].evidence =new bytes32[](0);
+		OpenCases[_id].status = 0;
 
 	}
-	function getClaim(bytes32 _claimId) returns (address plaintiff,bool valid,address beneficiary, uint256 blockNumber,
+	function getClaim(bytes32 _id) returns (address plaintiff,address beneficiary, uint256 blockNumber,
 			uint8 sig_v, bytes32 sig_r, bytes32 sig_s) {
-        return (OpenClaims[_claimId].plaintiff,
-                OpenClaims[_claimId].valid,
-                OpenClaims[_claimId].servicePromise.beneficiary,
-                OpenClaims[_claimId].servicePromise.blockNumber,
-                OpenClaims[_claimId].servicePromise.sig_v,
-                OpenClaims[_claimId].servicePromise.sig_r,
-                OpenClaims[_claimId].servicePromise.sig_s
+        return (OpenCases[_id].plaintiff,
+                OpenCases[_id].servicePromise.beneficiary,
+                OpenCases[_id].servicePromise.blockNumber,
+                OpenCases[_id].servicePromise.sig_v,
+                OpenCases[_id].servicePromise.sig_r,
+                OpenCases[_id].servicePromise.sig_s
                 );
     }
 
-  function setClaimValid(bytes32 _claimId)  {
-        OpenClaims[_claimId].valid = true;
-  }
-
 }
 
 contract SwearGame is Owned {
 
-	uint256 public amountStaked;
-	uint public rewardCompensation;
-	uint  public registeredPlayersCounter;
-	mapping(address => bool) public registeredPlayers;
-	mapping(address => bytes32[]) public clientsClaimsIds;
+	uint256 public deposit;
+	uint public reward;
+	uint  public playerCount;
+	mapping(address => bool) public players;
+	mapping(address => bytes32[]) public ids;
 
 	CaseContract public caseContract;
-	SampleToken public sampleToken;
-	bytes32 public claimId;
-
-
-	function SwearGame(address _caseContract, address _sampleToken, uint _rewardCompensation) {
-		caseContract = CaseContract(_caseContract);
-		sampleToken = SampleToken(_sampleToken);
-		rewardCompensation = _rewardCompensation;
-		registeredPlayersCounter = 0;
-
-	}
-
+	SampleToken public token;
+	bytes32 public id;
 
 
-
-	function deposit(uint256 _depositAmount) onlyOwner payable public returns(bool){
-
-		require(sampleToken.balanceOf(owner) >= _depositAmount);
-
-		require(sampleToken.transferFrom(owner, address(this), _depositAmount));
-
-		amountStaked += _depositAmount;
-
-		DepositStaked(_depositAmount, amountStaked);
-
-		return true;
+	function SwearGame(address _CaseContract, address _token, uint _reward) {
+		caseContract = CaseContract(_CaseContract);
+		token = SampleToken(_token);
+		reward = _reward;
+		playerCount = 0;
 
 	}
 
+  /* The function without name is the default function that is called whenever anyone sends funds to a contract */
+    function () payable {
+        uint amount = msg.value;
+        require(token.balanceOf(owner) >= amount);
+    		require(token.transferFrom(owner, address(this), amount));
```

Comment:
If this line returns false, it is in-effect doing the desired check as the line above, no?